beancount/parser/lexer.l

```diff
@@ -218,6 +222,10 @@ ACCOUNTNAME     ([A-Z0-9]|{UTF-8-ONLY})([A-Za-z0-9\-]|{UTF-8-ONLY})*
     return EOL;
 }
 
+<AFTERDATE>[ \t\r]+ {
+    BEGIN(AFTERDATE);
+}
```

Comment:
This looks funny and it seems to be required only because the `BEGIN(INITIAL)` added to `YY_USER_ACTION`. I think that getting rid of of both these would make the thing more understandable. I think the `BEGIN(INITIAL)` in `YY_USER_ACTION` can be placed in a default rule for the `<AFTERDATE>` state instead.